libuavcan_drivers/stm32/driver/src/uc_stm32_can.cpp

```diff
@@ -450,10 +450,35 @@ uavcan::int16_t CanIface::receive(uavcan::CanFrame& out_frame, uavcan::Monotonic
 uavcan::int16_t CanIface::configureFilters(const uavcan::CanFilterConfig* filter_configs,
                                            uavcan::uint16_t num_configs)
 {
-    // TODO: Hardware filter support
-    CriticalSectionLocker lock;
-    (void)filter_configs;
-    (void)num_configs;
+    if (num_configs < (NumFilters / 2)) {
+        CriticalSectionLocker lock;
+
+        can_->FMR |= bxcan::FMR_FINIT;
+
+        can_->FMR = (can_->FMR & ~0x00003F00) | static_cast<uint32_t>(NumFilters / 2) << 8; // Slave (CAN2) gets half of the filters
+
+        can_->FFA1R = 0x0AAAAAAA; // FIFO's are interleaved between filters
+        can_->FM1R = 0; // Identifier Mask mode
+        can_->FS1R = 0x7ffffff; // Single 32-bit for all
+
+        const uint8_t filter_start_index = (self_index_ == 0) ? 0 : NumFilters / 2;
+
+        for (uint8_t i = 0; i < (NumFilters / 2); i++) {
+            if (i < num_configs) {
+                can_->FilterRegister[filter_start_index + i].FR1 = filter_configs[i].id;
+                can_->FilterRegister[filter_start_index + i].FR2 = filter_configs[i].mask;
```

Comment:
The bit masks (EFF, RTR) should be converted from the UAVCAN format to the format used by bxCAN. There is also one horrible undocumented caveat I spent hours on, it is described in the STM32 driver from libcanard. Here: https://github.com/UAVCAN/libcanard/blob/master/drivers/stm32/canard_stm32.c#L449-L526

> If Mask asks us to accept both STDID and EXTID, we need to use EXT mode on the filter,
> otherwise it will reject all EXTID frames. This logic is not documented in the RM.